src/dmd/expressionsem.d

```diff
@@ -2150,13 +2151,33 @@ private bool functionParameters(const ref Loc loc, Scope* sc,
         (*arguments)[i] = arg;
     }
 
-    /* If calling C printf(), check the format string against the arguments
+    /* If calling C scanf(), sscanf(), fscanf(), printf(), sprintf() or fprintf(),
+       check the format string against the arguments
      */
-    if (tf.linkage == LINK.c && nparams >= 1 && fd && fd.ident == Id.printf)
+    if (tf.linkage == LINK.c && fd)
     {
-        if (auto se = (*arguments)[0].isStringExp())
+        if (fd.ident == Id.printf && nparams >= 1)
         {
-            if (checkPrintfFormat(se.loc, se.peekString(), (*arguments)[1 .. nargs]))
+            auto se = (*arguments)[0].isStringExp();
+            if (se && checkPrintfFormat(se.loc, se.peekString(), (*arguments)[1 .. nargs]))
+                err = true;
```

Comment:
the body of all of these are almost identical. Please write it as 
```d
int paramOffset = 0; // or some better name
if (fd.ident == Id.printf || fd.ident == Id.scanf)
    paramOffset = 1;
else if (fd.ident == Id.sprintf || fd.ident == Id.fprintf ||
            fd.ident == Id.sscanf || fd.ident == Id.fscanf)
    paramOffset = 2;
if (nparams >= paramOffset && paramOffset)
{
    auto se = (*arguments)[paramOffset-1].isStringExp();
    if (se && checkPrintfFormat(se.loc, se.peekString(), 
                                                    (*arguments)[paramOffset .. nargs]))
         err = true;
    }
}
```

Reply:
`printf` and `scanf` do not use the same function. Going to write something similar.